keps/0002-controller-manager.md

```diff
@@ -257,6 +263,102 @@ In case of the cloud-controller-manager, the deployment should be deleted using
 kubectl delete -f cloud-controller-manager.yml
 ```
 
+### Implementation Details/Notes/Constraints
+
+#### Repository Requirements
+
+**This is a proposed structure, and may change during the 1.11 release cycle.
+WG-Cloud-Provider will work with individual sigs to refine these requirements
+to maintain consistency while meeting the technical needs of the provider
+maintainers**
+
+Each cloud provider hosted within the `kubernetes` organization shall have a
+single repository named `kubernetes/cloud-provider-<provider_name>`. Those
+repositories shall have the following structure:
+
+* A `cloud-controller-manager` subdirectory that contains the implementation
+  of the provider-specific cloud controller.
+* A `docs` subdirectory.
+* A `docs/cloud-controller-manager.md` file that describes the options and
+  usage of the cloud controller manager code.
+* A `docs/testing.md` file that describes how the provider code is tested.
+* A `Makefile` with a `test` entrypoint to run the provider tests.
+
+Additionally, the repository should have:
+
+* A `docs/getting-started.md` file that describes the installation and basic
+  operation of the cloud controller manager code.
+
+Where the provider has additional capabilities, the repository should have
+the following subdirectories that contain the common features:
+
+* `dns` for DNS provider code.
+* `cni` for the Container Network Interface (CNI) driver.
+* `csi` for the Container Storage Interface (CSI) driver.
```

Comment:
What should be in put in `cni` or `csi` dir? Since their code are neutral to orchestrator, they should belong to a dedicate repo already ? `flex` makes sense here since it is for k8s only.

Reply:
Good point about CSI/CNI being neutral to orchestrators, I wouldn't be surprised if we remove these directories in the near future, I think today though it's useful since we can put deployment specs for running CSI/CNI on top of Kubernetes (daemonests, configmaps, etc).